tao_compiler/mlir/xla/ral/context/stream_executor_based_impl.cc

```diff
@@ -602,22 +602,91 @@ CudnnConvParamsKey makeConvTuningCacheKey(MemRefType<T, N>& input,
                                           MemRefType<int32_t, 1>& paddings,
                                           MemRefType<T, N>& output,
                                           MemRefType<int32_t, 1>& metadata) {
+  // Note metadata order:
+  //   - input layout: each field for one dimension. The order is:
+  //     * batch, channel, spatial dimensions
+  //   - kernel layout: each field for one dimension. The order is:
+  //     * in_channel, out_channel, spatial dimensions
+  //   - output layout: each field for one dimension. The order is:
+  //     * batch, channel, spatial dimensions
+  //   - strides: each filed for one spatial dimension.
+  //   - dilations: each filed for one spatial dimension.
   CudnnConvParamsKey key;
-  key.input_shape.reserve(N);
-  key.filter_shape.reserve(N);
-  key.output_shape.reserve(N);
-  for (int i = 0; i < N; ++i) {
-    key.input_shape.push_back(input.sizes[i]);
-    key.filter_shape.push_back(filter.sizes[i]);
-    key.output_shape.push_back(output.sizes[i]);
+  static_assert(N > 2);
+  const int effectiveN = (N < 4) ? 4 : N;
+  std::vector<int64_t> input_sizes(effectiveN);
+  std::vector<int64_t> filter_sizes(effectiveN);
+  std::vector<int64_t> output_sizes(effectiveN);
+  int64_t padding_orig_sizes = paddings.sizes[0];
+  std::vector<int32_t> padding_data(effectiveN);
+  int64_t metadata_orig_sizes = metadata.sizes[0];
+  std::vector<int32_t> metadata_data;
+
+  bool is_1d_conv = (N == 3);
+  // input, kernel, and output layout & sizes.
+  for (int64_t layout_idx = 0; layout_idx < 3; layout_idx++) {
```

Comment:
Is that meant we use conv2d to imitate conv1d?

Reply:
Yes, you are right.